### PR TITLE
[16732] on_unacknowledged_sample_removed documentation

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
This PR includes documentation of the new DataWriterListener callback `on_unacknowledged_sample_removed`. It should be merged once the implementation has been done in Fast DDS.

Signed-off-by: JLBuenoLopez-eProsima <joseluisbueno@eprosima.com>

### CODE REVIEW

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]

File: code/DDSCodeTester.cpp [hunk 1]
```diff
@@ -2406,10 +2422,20 @@ class CustomDataWriterListener : public DataWriterListener
...
     {
-        (void)writer, (void)status;
+        static_cast<void>(writer);
+        static_cast<void>(status);
         std::cout << "Liveliness lost. Matched Subscribers will consider us offline" << std::endl;
     }
 
+    virtual void on_unacknowledged_sample_removed(
+            DataWriter* writer,
+            const InstanceHandle_t& instance)
```
Comment (MiguelCompany) [inline code/DDSCodeTester.cpp:2432]: I would like the CI to fail if we break the API in the future...

```suggestion
    void on_unacknowledged_sample_removed(
            DataWriter* writer,
            const InstanceHandle_t& instance) override
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Comment by MiguelCompany** [comment]
New manual CI [here](http://jenkins.eprosima.com:8080/job/fastdds_docs_manual/636)

**Comment by JLBuenoLopez-eProsima** [comment]
[Another manual CI](http://ci.intranet.eprosima.com:8080/job/fastdds_docs_manual/640/)

**Pull request merged**